robot/client/client.go

```diff
@@ -278,6 +279,7 @@ func New(ctx context.Context, address string, logger golog.Logger, opts ...Robot
 	// interceptors are applied in order from first to last
 	rc.dialOptions = append(
 		rc.dialOptions,
+		rpc.WithUnaryClientInterceptor(rdkutils.ContextWithMetadataUnaryClientInterceptor),
```

Comment:
@edaniels Do we need to register interceptors anywhere else? This is the only place that seemed necessary for this to work.